src/coreComponents/mesh/generators/MeshGeneratorBase.hpp

```diff
@@ -78,13 +79,13 @@ class MeshGeneratorBase : public dataRepository::Group
   virtual void generateMesh( DomainPartition & domain ) = 0;
 
   /**
-   * @brief Import data from external sources (e.g. dataset that comes with a mesh).
-   * @param[in] domain the domain partition
+   * @brief import fields from the mesh  on the array accessible via the given wrapper.
+   * @param cellBlockName name of the cell block to copy data from.
+   * @param meshFieldName name of the field in the meshd
+   * @param isMaterialField Indicate if we want to import material or regular fields
+   * @param wrapper Wrapper to access the array
    */
-  virtual void importFields( DomainPartition & domain ) const
-  {
-    GEOSX_UNUSED_VAR( domain );
-  }
+  virtual void importFieldsOnArray( string const cellBlockName, string const meshFieldName, bool isMaterialField, dataRepository::WrapperBase & wrapper ) const = 0;
```

Comment:
```suggestion
  virtual void importFieldsOnArray( string const & cellBlockName, string const & meshFieldName, bool isMaterialField, dataRepository::WrapperBase & wrapper ) const = 0;
```
It's not necessary to copy as const a string, since it'll be copied.
Even if at some place, we do define copied input as `const`, so we can't modify the input tmp param...